substra/sdk/backends/local/backend.py

```diff
@@ -739,8 +736,309 @@ def describe(self, asset_type, key):
         with open(asset.description, "r", encoding="utf-8") as f:
             return f.read()
 
-    def leaderboard(self, objective_key, sort="desc"):
-        raise NotImplementedError
+    def __format_for_leaderboard(self, testtuple):
+        for tuple_type in [
+                schemas.Type.Traintuple,
+                schemas.Type.CompositeTraintuple,
+                schemas.Type.Aggregatetuple
+        ]:
+            try:
+                traintuple = self._db.get(tuple_type, testtuple.traintuple_key)
+                traintuple_type = tuple_type
+                break
+            except exceptions.NotFound:
+                pass
+        if traintuple_type == schemas.Type.Traintuple:
+            algo_type = schemas.Type.Algo
+        elif traintuple_type == schemas.Type.Aggregatetuple:
+            algo_type = schemas.Type.AggregateAlgo
+        elif traintuple_type == schemas.Type.CompositeTraintuple:
+            algo_type = schemas.Type.CompositeAlgo
+
+        algo = self._db.get(algo_type, traintuple.algo_key)
```

Comment:
This could probably be simplified a bit with the introduction of the algo_type attributes on the tuples.

Reply:
Indeed !